src/main/java/de/telran/processor/services/FileService.java

```diff
@@ -10,7 +10,28 @@
 
 public class FileService {
     public List<ImageDescriptor> readImageDescriptions(String fileName) {
-        return new ArrayList<>();
+        List<String> listOfImageDescriptions = readFileToStringList(fileName);
+        return listOfImageDescriptions.stream().map(this::convertStringToObject).collect(Collectors.toList());
+    }
+    private ImageDescriptor convertStringToObject(String line){
+        return new ImageDescriptor(line.split("::")[0], line.split("::")[1]);
+    }
+
+    private List<String> readFileToStringList(String fileName) {
+        List<String> stringList = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            stringList = bufferedReader.lines().collect(Collectors.toList());
```

Comment:
You could use a method reference to `convertStringToObject() ` in the Stream.map() method in order to map read strings to ImageDescriptor objects, so you save one traverse over the list in your readImageDescriptions() method.